specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/automations.json

```diff
@@ -0,0 +1,690 @@
+              }
+            ]
+          }
+        },
+        "ruleSets": {
+          "type": "array",
+          "description": "A set of rules which evaluate upon event interception. A logical disjunction is applied between defined rule sets (logical �or�).",
+          "items": {
+            "$ref": "#/definitions/AutomationRuleSet"
+          }
+        }
+      }
+    },
+    "AutomationScope": {
+      "type": "object",
+      "description": "A single automation scope.",
+      "properties": {
+        "description": {
+          "type": "string",
+          "description": "The resources scope description."
+        },
+        "scopePath": {
+          "type": "string",
+          "description": "The resources scope path. Can be the subscription on which the automation is defined on or a resource group under that subscription (fully qualified Azure resource IDs)."
+        }
+      }
+    },
+    "AutomationAction": {
+      "type": "object",
+      "description": "The action that should be triggered.",
+      "discriminator": "actionType",
+      "properties": {
+        "actionType": {
+          "type": "string",
+          "description": "The type of the action that will be triggered by the Automation",
+          "enum": [ "LogicApp", "EventHub", "SecurityEmail", "Workspace" ],
+          "x-ms-enum": {
+            "name": "ActionType",
+            "modelAsString": true,
+            "values": [
+              {
+                "value": "LogicApp"
+              },
+              {
+                "value": "EventHub"
+              },
+              {
+                "value": "SecurityEmail"
+              },
+              {
+                "value": "Workspace"
+              }
+            ]
+          }
+        }
+      }
+    },
+    "AutomationActionLogicApp": {
+      "type": "object",
+      "description": "The logic app action that should be triggered.",
+      "x-ms-discriminator-value": "LogicApp",
+      "allOf": [
+        {
+          "$ref": "#/definitions/AutomationAction"
+        }
+      ],
+      "properties": {
+        "logicAppResourceId": {
+          "type": "string",
+          "description": "The triggered Logic App Azure Resource ID. This can also reside on other subscriptions, given that you have permissions to trigger the Logic App"
+        },
+        "uri": {
+          "type": "string",
+          "description": "The Logic App trigger URI endpoint.",
+          "format": "uri"
+        }
+      }
+    },
+    "AutomationActionEventHub": {
+      "type": "object",
+      "description": "The target Event Hub to which event data will be exported.",
+      "x-ms-discriminator-value": "EventHub",
+      "allOf": [
+        {
+          "$ref": "#/definitions/AutomationAction"
+        }
+      ],
+      "properties": {
+        "eventHubResourceId": {
+          "type": "string",
+          "description": "The Event Hub Azure Resource ID."
+        },
+        "connectionString": {
```

Comment:
Does this include creds? If so, please confirm you're not going to return it in a GET.

Reply:
Thanks for noticing it, the connection string will not be returned in GET

Reply:
@anthony-c-martin , in a second thought - why can't they be returned back to the user? they are provided by the user. Appreciate more info and what should be the best practice here